File: WarehouseWebsite.Domain/Models/Orders/Order.cs
using WarehouseWebsite.Domain.Models.Customers;

namespace WarehouseWebsite.Domain.Models.Orders
{
    public class Order : BaseEntity
    {
        public Guid CustomerId { get; set; }
        public DateTime OrderTime { get; set; }
        public OrderStatus Status { get; set; }

        public decimal TotalPrice { get; set; }

        public Customer Customer { get; set; } = null!;
        public IEnumerable<OrderItem> OrderItems { get; set; } = new List<OrderItem>();
    }

    public class AwaitingOrder : Order 
    {
        public AwaitingOrder() { }

        public AwaitingOrder(Order order)
        {
            CustomerId = order.CustomerId;
            OrderTime = order.OrderTime;
            Status = OrderStatus.Awaiting;
            TotalPrice = order.TotalPrice;
            Customer = order.Customer;
            OrderItems = order.OrderItems;
        }

        public Order ToOrder()
        {
            return new Order
            {
                CustomerId = CustomerId,
                OrderTime = OrderTime,
                Status = OrderStatus.Transiting,
                TotalPrice = TotalPrice,
                Customer = Customer,
                OrderItems = OrderItems,
            };
        }
    }
}

File: WarehouseWebsite.Frontend/src/app/shopping/order/order.ts
export class Order {
  id: string;
  customerId: string;
  orderTime: string;
  status: string;
  totalPrice: number;
  orderItems: OrderItem[];

  constructor(id: string, customerId: string, orderTime: string, status: string, totalPrice: number, orderItems: OrderItem[]) {
    this.id = id;
    this.customerId = customerId;
    this.orderTime = orderTime;
    this.status = status;
    this.totalPrice = totalPrice;
    this.orderItems = orderItems;
  }
}

export class OrderItem {
  itemId: string;
  quantity: number;
  price: number;

  constructor(itemId: string, quantity: number, price: number) {
    this.itemId = itemId;
    this.quantity = quantity;
    this.price = price;
  }
}
